platform/android/display_server_android.cpp

```diff
@@ -554,30 +564,47 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
 	}
 #endif
 
-#if defined(VULKAN_ENABLED)
-	context_vulkan = nullptr;
-	rendering_device_vulkan = nullptr;
+#if defined(RD_ENABLED)
+	context_rd = nullptr;
+	rendering_device = nullptr;
+
+	ANativeWindow *native_window = OS_Android::get_singleton()->get_native_window();
+	ERR_FAIL_NULL(native_window);
 
+#if defined(VULKAN_ENABLED)
 	if (rendering_driver == "vulkan") {
-		ANativeWindow *native_window = OS_Android::get_singleton()->get_native_window();
-		ERR_FAIL_NULL(native_window);
```

Comment:
This change seems to be the source of my problems when using the Compatibility renderer. Previously, the `ERR_FAIL_NULL(native_window);` would only happen when using Vulkan, but this PR causes it to happen always.

Applying this diff fixes it for me:

```
diff --git a/platform/android/display_server_android.cpp b/platform/android/display_server_android.cpp
index a133cd3ee2..640441a4b3 100644
--- a/platform/android/display_server_android.cpp
+++ b/platform/android/display_server_android.cpp
@@ -568,9 +568,6 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
        context_rd = nullptr;
        rendering_device = nullptr;
 
-       ANativeWindow *native_window = OS_Android::get_singleton()->get_native_window();
-       ERR_FAIL_NULL(native_window);
-
 #if defined(VULKAN_ENABLED)
        if (rendering_driver == "vulkan") {
                context_rd = memnew(VulkanContextAndroid);
@@ -578,6 +575,9 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
 #endif
 
        if (context_rd) {
+               ANativeWindow *native_window = OS_Android::get_singleton()->get_native_window();
+               ERR_FAIL_NULL(native_window);
+
                if (context_rd->initialize() != OK) {
                        memdelete(context_rd);
                        context_rd = nullptr;
```

Reply:
Makes sense! The fix will come in the imminent next push.